File: db4oj/core/src/com/db4o/internal/ReferenceSystem.java
/* Copyright (C) 2007  db4objects Inc.  http://www.db4o.com */

package com.db4o.internal;

import com.db4o.foundation.*;

/**
 * @exclude
 */
public interface ReferenceSystem {

	public void addNewReference(ObjectReference ref);

	public void addExistingReference(ObjectReference ref);

	public void addExistingReferenceToObjectTree(ObjectReference ref);

	public void addExistingReferenceToIdTree(ObjectReference ref);
	
	public void commit();
	
	public ObjectReference referenceForId(int id);

	public ObjectReference referenceForObject(Object obj);

	public void removeReference(ObjectReference ref);
	
	public void rollback();
	
	public void traverseReferences(Visitor4 visitor);

}

File: db4on/core/src/core/com/db4o/internal/ReferenceSystem.cs
namespace com.db4o.@internal
{
	/// <exclude></exclude>
	public interface ReferenceSystem
	{
		void AddNewReference(com.db4o.@internal.ObjectReference @ref);

		void AddExistingReference(com.db4o.@internal.ObjectReference @ref);

		void AddExistingReferenceToObjectTree(com.db4o.@internal.ObjectReference @ref);

		void AddExistingReferenceToIdTree(com.db4o.@internal.ObjectReference @ref);

		void Commit();

		com.db4o.@internal.ObjectReference ReferenceForId(int id);

		com.db4o.@internal.ObjectReference ReferenceForObject(object obj);

		void RemoveReference(com.db4o.@internal.ObjectReference @ref);

		void Rollback();

		void TraverseReferences(com.db4o.foundation.Visitor4 visitor);
	}
}
